web/src/types.ts

```diff
@@ -6,13 +6,67 @@ export type MatchedMessage = {
 
 // TODO/typescript: Move this to message_store
 export type MessageType = "private" | "stream";
+export type MessageContentType = "text/html" | "text/x-markdown";
+export type MessageReactionType = "unicode_emoji" | "realm_emoji" | "zulip_extra_emoji";
+
+// TODO/typescript: Move these types to message_store
+
+export type DisplayRecipient = {
+    email: string;
+    full_name: string;
+    id: number;
+    is_mirror_dummy: boolean;
+};
+
+export type MessageEditHistory = {
```

Comment:
This should be called `MessageEditHistoryEntry` or `MessageEditHistoryEvent`, since these elements are individual entries in the edit history.